app/export.go

```diff
@@ -120,19 +148,146 @@ func (a *App) BulkExport(ctx request.CTX, writer io.Writer, outPath string, opts
 	}
 
 	if opts.IncludeAttachments {
-		ctx.Logger().Info("Bulk export: exporting file attachments")
+		mlog.Info("Bulk export: exporting file attachments")
+		exportFilename := a.GetId() + "_export.log"
 		for _, attachment := range attachments {
+			isExist, err := a.FileExists(*attachment.Path)
+			if err != nil {
+				return err
+			}
+			if !isExist {
+				f, err := os.OpenFile(filepath.Join("./data/export/", exportFilename), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+				if err != nil {
+					return model.NewAppError("openFile", "Failed to open file", nil, err.Error(), http.StatusInternalServerError)
+				}
+				defer f.Close()
+				ctime, err := strconv.ParseInt(strconv.FormatInt(attachment.CreateAt, 10), 10, 64)
+				if err != nil {
+					return model.NewAppError("parseTime", "Failed to parse create at time", nil, err.Error(), http.StatusInternalServerError)
+				}
+				utime, err := strconv.ParseInt(strconv.FormatInt(attachment.UpdateAt, 10), 10, 64)
+				if err != nil {
+					return model.NewAppError("parseTime", "Failed to parse update at time", nil, err.Error(), http.StatusInternalServerError)
+				}
+				create_time := time.UnixMilli(ctime)
+				update_time := time.UnixMilli(utime)
+				data := LogFormat{
+					Id:            attachment.Id,
+					CreatorId:     attachment.CreatorId,
+					PostId:        attachment.PostId,
+					ChannelId:     attachment.ChannelId,
+					CreateAt:      create_time,
+					UpdateAt:      update_time,
+					Path:          *attachment.Path,
+					ThumbnailPath: attachment.ThumbnailPath,
+					PreviewPath:   attachment.PreviewPath,
+					Name:          attachment.Name,
+					Extension:     attachment.Extension,
+					Size:          attachment.Size,
+					MimeType:      attachment.MimeType,
+					Width:         attachment.Width,
+					Height:        attachment.Height,
+				}
+				file, _ := json.MarshalIndent(data, "", " ")
+
+				if _, err := f.WriteString(string(file) + "\n"); err != nil {
+					return model.NewAppError("exportLogFile", "Failed to export the missing information of attachments in log file", nil, err.Error(), http.StatusInternalServerError)
+				}
+				mlog.Warn(fmt.Sprintf("attachment %v not found and will not be exported", *attachment.Path))
+				continue
+			}
 			if err := a.exportFile(outPath, *attachment.Path, zipWr); err != nil {
 				return err
 			}
 		}
 		for _, attachment := range directAttachments {
+			isExist, err := a.FileExists(*attachment.Path)
+			if err != nil {
+				return err
+			}
+			if !isExist {
```

Comment:
this code is looking similar to the above code if it is make a function for this and use that everywhere

Reply:
Okk